### PR TITLE
Add Testing section to the PR template

### PR DESCRIPTION
**Pull request by NiseVoid**
# Objective

- Reviews could go smoother if reviewers know how a PR was tested, how they can test it, and if it needs more testing.

## Solution

- Add a testing section

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -7,6 +7,12 @@
 
 - Describe the solution used to achieve the objective above.
 
+## Testing
+
+- Did you test these changes? If so, how have you tested them? Are there any parts that need more testing?
+- Is there anything people need to know when testing these changes?
```
Comment (alice-i-cecile) [inline .github/pull_request_template.md:13]:
```suggestion
- Is there anything people need to know when manually testing these changes?
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -7,6 +7,12 @@
 
 - Describe the solution used to achieve the objective above.
 
+## Testing
+
+- Did you test these changes? If so, how have you tested them? Are there any parts that need more testing?
```
Comment (alice-i-cecile) [inline .github/pull_request_template.md:12]:
```suggestion
- Did you perform any manual tests for these changes? If so, how?
- Are there any parts that need more manual testing?
```

File: .github/pull_request_template.md (hunk 1 above)
Comment (inodentry) [inline .github/pull_request_template.md:13]:
```suggestion
- How can other people (reviewers) test your changes? Is there anything specific they need to know?
```

**Comment by inodentry** [comment]
@alice-i-cecile why the focus on "manual"/"manually"?

**Comment by alice-i-cecile** [comment]
Automated testing is extremely valuable and should be preferred. I was thinking this would maybe be clearer, but perhaps there's a better way to encourage that.

**Comment by inodentry** [comment]
I think the main goals of having this in the pr template should be to encourage random people from the community to test PRs, also to make it easier for reviewers, and to make sure that PR authors communicate about how confident they feel that their stuff works.

Whether it is automatic or manual testing is a secondary concern. I think it is better to omit such wording. By asking "what have you done to test this and how can others test it", PR authors will naturally think to mention both automatic and manual forms of testing, as relevant.

**Review by mgi388 — APPROVED** [review]
This echoes the sentiment I had in my [Discord post] a little while ago as a new reviewer wanting to know how to test a PR, or wanting to gain more confidence that the author has put some thought into testing.

Leaving an approval, FWIW.

[Discord post]: https://discord.com/channels/691052431525675048/692572690833473578/1227785809910956133

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**